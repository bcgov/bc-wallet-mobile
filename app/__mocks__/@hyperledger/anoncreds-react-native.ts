const AnonCredsReactNative = {
  install: jest.fn(),
}

export default AnonCredsReactNative
