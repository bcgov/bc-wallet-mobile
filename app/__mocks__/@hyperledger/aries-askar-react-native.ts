const AriesAskarReactNative = {
  install: jest.fn(),
}

export default AriesAskarReactNative
