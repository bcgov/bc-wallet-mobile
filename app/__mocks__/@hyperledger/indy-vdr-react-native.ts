const IndyVDRReactNative = {
  install: jest.fn(),
}

export default IndyVDRReactNative
