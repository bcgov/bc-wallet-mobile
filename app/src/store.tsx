import AsyncStorage from '@react-native-async-storage/async-storage'
import {
  State as BifoldState,
  mergeReducers,
  reducer as bifoldReducer,
  defaultState,
  ReducerAction,
} from 'aries-bifold'

export interface IASEnvironment {
  name: string
  iasAgentInviteUrl: string
  iasPortalUrl: string
}
interface Developer {
  environment: IASEnvironment
}

interface DismissPersonCredentialOffer {
  personCredentialOfferDismissed: boolean
}

export interface BCState extends BifoldState {
  developer: Developer
  dismissPersonCredentialOffer: DismissPersonCredentialOffer
}

enum DeveloperDispatchAction {
  UPDATE_ENVIRONMENT = 'developer/updateEnvironment',
}

enum DismissPersonCredentialOfferDispatchAction {
  PERSON_CREDENTIAL_OFFER_DISMISSED = 'dismissPersonCredentialOffer/personCredentialOfferDismissed',
}

export type BCDispatchAction = DeveloperDispatchAction | DismissPersonCredentialOfferDispatchAction

export const BCDispatchAction = {
  ...DeveloperDispatchAction,
  ...DismissPersonCredentialOfferDispatchAction,
}

export const iasEnvironments: Array<IASEnvironment> = [
  {
    name: 'Production',
    iasAgentInviteUrl:
      'https://idim-agent.apps.silver.devops.gov.bc.ca?c_i=eyJAdHlwZSI6ICJkaWQ6c292OkJ6Q2JzTlloTXJqSGlxWkRUVUFTSGc7c3BlYy9jb25uZWN0aW9ucy8xLjAvaW52aXRhdGlvbiIsICJAaWQiOiAiZTZiY2EwNzQtYmNmNC00ZjQzLTgwMjYtNWNjZjhkN2M4OTQyIiwgImxhYmVsIjogIklESU0iLCAic2VydmljZUVuZHBvaW50IjogImh0dHBzOi8vaWRpbS1hZ2VudC5hcHBzLnNpbHZlci5kZXZvcHMuZ292LmJjLmNhIiwgImltYWdlVXJsIjogImh0dHBzOi8vaWQuZ292LmJjLmNhL3N0YXRpYy9Hb3YtMi4wL2ltYWdlcy9mYXZpY29uLmljbyIsICJyZWNpcGllbnRLZXlzIjogWyJHeXJxY2NQR1FtV3JxWGZ6cWtRTUF2VlBycmRWYUdwRkgxOHNOWkUzdUtGIl19',
    iasPortalUrl: 'https://id.gov.bc.ca/issuer/v1/dids',
  },
  {
    name: 'Development',
    iasAgentInviteUrl:
      'https://idim-agent-dev.apps.silver.devops.gov.bc.ca?c_i=eyJAdHlwZSI6ICJkaWQ6c292OkJ6Q2JzTlloTXJqSGlxWkRUVUFTSGc7c3BlYy9jb25uZWN0aW9ucy8xLjAvaW52aXRhdGlvbiIsICJAaWQiOiAiZDQ3NWM3ZjQtMTRjMy00NzdkLWI2NTMtY2Y5MDM4NDJmNGJjIiwgInJlY2lwaWVudEtleXMiOiBbIjJlSHBRRm9uTUVobXpvYWVFbUFMS1dxZHF5UldZZkE2TFF5akpKdGdiMldUIl0sICJsYWJlbCI6ICJJRElNIChEZXYpIiwgInNlcnZpY2VFbmRwb2ludCI6ICJodHRwczovL2lkaW0tYWdlbnQtZGV2LmFwcHMuc2lsdmVyLmRldm9wcy5nb3YuYmMuY2EiLCAiaW1hZ2VVcmwiOiAiaHR0cHM6Ly9pZC5nb3YuYmMuY2Evc3RhdGljL0dvdi0yLjAvaW1hZ2VzL2Zhdmljb24uaWNvIn0=',
    iasPortalUrl: 'https://iddev.gov.bc.ca/issuer/v1/dids',
  },
  {
    name: 'Test',
    iasAgentInviteUrl:
      'https://idim-sit-agent-dev.apps.silver.devops.gov.bc.ca?c_i=eyJAdHlwZSI6ICJkaWQ6c292OkJ6Q2JzTlloTXJqSGlxWkRUVUFTSGc7c3BlYy9jb25uZWN0aW9ucy8xLjAvaW52aXRhdGlvbiIsICJAaWQiOiAiNmNjMjJiNTYtZmQwYy00Yjc4LWE3ZTQtYzYwYzJlODBlMDM0IiwgInJlY2lwaWVudEtleXMiOiBbIkNoSmJDTTVZSlMxb3hTQU1WNU1vY1J5cE1tUVp0eFFqcG9KWEZpTHZnMUM5Il0sICJsYWJlbCI6ICJJRElNIChTSVQpIiwgInNlcnZpY2VFbmRwb2ludCI6ICJodHRwczovL2lkaW0tc2l0LWFnZW50LWRldi5hcHBzLnNpbHZlci5kZXZvcHMuZ292LmJjLmNhIiwgImltYWdlVXJsIjogImh0dHBzOi8vaWQuZ292LmJjLmNhL3N0YXRpYy9Hb3YtMi4wL2ltYWdlcy9mYXZpY29uLmljbyJ9',
    iasPortalUrl: 'https://idsit.gov.bc.ca/issuer/v1/dids',
  },
]

const developerState: Developer = {
  environment: iasEnvironments[0],
}

const dismissPersonCredentialOfferState: DismissPersonCredentialOffer = {
  personCredentialOfferDismissed: false,
}

export enum BCLocalStorageKeys {
  PersonCredentialOfferDismissed = 'PersonCredentialOfferDismissed',
  Environment = 'Environment',
}

export const initialState: BCState = {
  ...defaultState,
  developer: developerState,
  dismissPersonCredentialOffer: dismissPersonCredentialOfferState,
}

const bcReducer = (state: BCState, action: ReducerAction<BCDispatchAction>): BCState => {
  switch (action.type) {
    case DeveloperDispatchAction.UPDATE_ENVIRONMENT: {
      const environment: IASEnvironment = (action?.payload || []).pop()
      const developer = { ...state.developer, environment }

      // Persist IAS environment between app restarts
      AsyncStorage.setItem(BCLocalStorageKeys.Environment, JSON.stringify(developer.environment))
      return { ...state, developer }
    }
    case DismissPersonCredentialOfferDispatchAction.PERSON_CREDENTIAL_OFFER_DISMISSED: {
      const { personCredentialOfferDismissed } = (action?.payload || []).pop()
      const dismissPersonCredentialOffer = { ...state.dismissPersonCredentialOffer, personCredentialOfferDismissed }
      const newState = { ...state, dismissPersonCredentialOffer }

      // save to storage so notification doesn't reapper on app restart
      AsyncStorage.setItem(
        BCLocalStorageKeys.PersonCredentialOfferDismissed,
        JSON.stringify(newState.dismissPersonCredentialOffer)
      )
      return newState
    }
    default:
      return state
  }
}

// @ts-ignore
export const reducer = mergeReducers(bifoldReducer, bcReducer)
