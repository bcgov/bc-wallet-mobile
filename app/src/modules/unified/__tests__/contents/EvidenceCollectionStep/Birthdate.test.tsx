import { testIdWithKey, StoreProvider } from '@hyperledger/aries-bifold-core'
import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { initialState, reducer } from '../../../../../store'
import BirthdateContent from '../../../contents/EvidenceCollectionStep/Birthdate'

describe('BirthdateContent Component', () => {
  const onComplete = jest.fn()

  beforeAll(() => {
    jest.useFakeTimers()
    // Set the date to 20 Oct 2024 00:12:00 GMT so snapshots don't change
    jest.setSystemTime(new Date('20 Oct 2024 00:12:00 GMT').getTime())
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  test('renders correctly', () => {
    const tree = render(
      <StoreProvider initialState={initialState} reducer={reducer}>
        <BirthdateContent onComplete={onComplete} />
      </StoreProvider>
    )
    expect(tree).toMatchSnapshot()
  })

  test('Done test ID is present and press handler works', () => {
    const { getByTestId } = render(
      <StoreProvider initialState={initialState} reducer={reducer}>
        <BirthdateContent onComplete={onComplete} />
      </StoreProvider>
    )

    const doneButton = getByTestId(testIdWithKey('Done'))
    expect(doneButton).toBeDefined()
    fireEvent(doneButton, 'press')
    expect(onComplete).toHaveBeenCalledTimes(1)
  })
})
