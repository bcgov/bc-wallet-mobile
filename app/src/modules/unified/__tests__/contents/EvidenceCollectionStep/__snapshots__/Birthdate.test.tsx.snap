// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirthdateContent Component renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
    ]
  }
  style={
    {
      "backgroundColor": "#313132",
      "height": "100%",
      "justifyContent": "space-between",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flex": 1,
        "padding": 24,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
            "marginBottom": 8,
          }
        }
      >
        Unified.Birthdate.CardSerialNumber
      </Text>
      <View
        style={
          {
            "backgroundColor": "#F2F2F2",
            "height": 8,
            "marginBottom": 16,
            "width": "100%",
          }
        }
      />
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 26,
            "fontWeight": "bold",
            "marginBottom": 16,
          }
        }
      >
        Unified.Birthdate.Heading
      </Text>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
            "marginBottom": 8,
          }
        }
      >
        Unified.Birthdate.Paragraph
      </Text>
      <RNDatePicker
        cancelText="Cancel"
        confirmText="Confirm"
        date={1729383120000}
        dividerColor="#000000"
        maximumDate={1729383120000}
        minuteInterval={1}
        mode="date"
        onChange={[Function]}
        onDateChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "height": 216,
              "width": 310,
            },
            undefined,
          ]
        }
        textColor="#000000"
        theme="light"
        timeZoneOffsetInMinutes=""
        title="Select date"
      />
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": "auto",
        "position": "relative",
      }
    }
  >
    <View
      accessibilityLabel="Global.Done"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/Done"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
              false,
              false,
            ]
          }
        >
          Global.Done
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
